### PR TITLE
nova,neutron,glance: keystone auth_token memcached protection

### PR DESCRIPTION
**Pull request by stefannica**
This PR continues the work done in #1090 by configuring the auth_token keystone middleware for all services currently using memcached caching (neutron, glance, nova) to authenticate and encrypt the token data stored in memcached using a generated secret key.
    
More information: 
- https://docs.openstack.org/keystonemiddleware/latest/middlewarearchitecture.html#memcache-protection

### CODE REVIEW

File: chef/data_bags/crowbar/migrate/glance/201_add_memcache_secret_key.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+def upgrade(ta, td, a, d)
+  if a["memcache_secret_key"].nil? || a["memcache_secret_key"].empty?
+    service = ServiceObject.new "fake-logger"
+    a["memcache_secret_key"] = service.random_password
+  end
+  return a, d
+end
+
+def downgrade(ta, td, a, d)
+  a.delete["memcache_secret_key"]
```
Comment (cmurphy) [inline chef/data_bags/crowbar/migrate/glance/201_add_memcache_secret_key.rb]: This needs to be `a.delete("memcache_secret_key")` (parentheses not brackets)

Comment (stefannica) [inline chef/data_bags/crowbar/migrate/glance/201_add_memcache_secret_key.rb]: ooops !  good eye @cmurphy ! thanks !

File: chef/data_bags/crowbar/template-neutron.json [hunk 2]
```diff
@@ -172,7 +173,7 @@
     "neutron": {
       "crowbar-revision": 0,
       "crowbar-applied": false,
-      "schema-revision": 200,
+      "schema-revision": 201,
```
Comment (cmurphy) [inline chef/data_bags/crowbar/template-neutron.json]: This needs to be bumped

**Comment by stefannica** [comment]
HA gating job fail is fixed by #1125

**Comment by stefannica** [comment]
@cmurphy , @sayalilunkad would you please recheck this PR ? All I did since your last review was I rebased a couple of times and updated the schema version numbers to stay on top of master.

**Pull request merged**